File: 3168/3168.go
func minimumNumberOfChairsInAWaitin(root *TreeNode) int {
	if root == nil {
		return 0
	}
	l := minimumNumberOfChairsInAWaitin(root.Left)
	r := minimumNumberOfChairsInAWaitin(root.Right)
	if l > r {
		return 1 + l
	}
	return 1 + r
}

File: 3168/3168.js
/**
 * @param {TreeNode} root
 * @return {number}
 */
var minimumNumberOfChairsInAWaitin = function(root) {
    if (!root) return 0;
    return 1 + Math.max(minimumNumberOfChairsInAWaitin(root.left), minimumNumberOfChairsInAWaitin(root.right));
};